lib/ClangImporter/ClangImporter.cpp

```diff
@@ -7717,26 +7722,44 @@ ValueDecl *
 CxxRecordAsSwiftType::evaluate(Evaluator &evaluator,
                                CxxRecordSemanticsDescriptor desc) const {
   auto cxxDecl = dyn_cast<clang::CXXRecordDecl>(desc.decl);
-  if (!cxxDecl)
+  // TODO: support specializations like swift::Optional<int>
+  if (!cxxDecl || isa<clang::ClassTemplateSpecializationDecl>(cxxDecl))
```

Comment:
Do you need the template specialization check here?  `Optional` gets ruled out by your storage check below, right?

Reply:
True, the storage check rules out `swift::Optional` as of today. But not necessarily in the future, if/when `swift::Optional<TrivialType>` has a different representation.